### PR TITLE
DTGB-491: Added theming for dg_maps

### PR DESCRIPTION
**Pull request by hstandaert**
## Description

Added theming for the output for the `dg_maps` module.

### Remarks
**Javascript**
Some of the JS behaviors for dg_maps had to be overridden and resulted in some code that was duplicated and slightly adapted (since we don't have any other option to do this)

**Theming**
Since we don't control the markup for the maps, we have to scope the css to the provided selectors

### Trivial
- Added `js:watch` task to gent-base gulpfile

## How Has This Been Tested?

- Local environment
- In multiple browsers

## Screenshots (if appropriate):
screenshot 2019-01-22 at 14 11 12

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have updated the documentation accordingly.
- [ ] I have updated the style guide CHANGELOG accordingly.
- [x] I have updated the gent_base CHANGELOG accordingly.
- [x] I have ran gulp axe on the compiled files and found no errors.

### CODE REVIEW

File: gent_base.libraries.yml [hunk 1]
```diff
@@ -80,3 +80,12 @@ opening_hours:
...
     - gent_base/accordion
     - gent_base/tabs
+
+maps:
+  version: 1.x
+  js:
+    source/js/dg_maps.functions.js: { minified: false }
+  dependencies:
+    - gent_base/accordion
+    - gent_base/accordion
```
Comment (delrueba) [inline gent_base.libraries.yml]: Why is this double?

File: source/gulpfile.js [hunk 2]
```diff
@@ -112,6 +121,8 @@ gulp.task('build', gulp.parallel('validate', 'compile'));
  *  gulp watch
  *
  * Used for local development to compile and validate after every change.
- *
+ *;
```
Comment (delrueba) [inline source/gulpfile.js]:
```suggestion
 *
```

File: source/gulpfile.js [hunk 3]
```diff
@@ -112,6 +121,8 @@ gulp.task('build', gulp.parallel('validate', 'compile'));
...
  */
-gulp.task('default', gulp.parallel('validate', 'compile'));
+gulp.task('default', gulp.series('js:watch'));
+gulp.task('watch', gulp.series('default'))
```
Comment (delrueba) [inline source/gulpfile.js]:
```suggestion
gulp.task('watch', gulp.series('default'));
```

File: source/js/dg_maps.functions.js [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+  Drupal.dgMaps.ol.control.DataLayerSwitcher = function (opt_options) {
+    var options = opt_options || {};
+
+    /**
+     * The categories definitions.
+     *
+     * @private
+     * @type {Object}
+     */
+    this.categories_ = typeof options.categories !== 'undefined' ? options.categories : [];
```
Comment (delrueba) [inline source/js/dg_maps.functions.js:26]:
```suggestion
    this.categories_ = options.categories || [];
```

Comment (delrueba) [reply source/js/dg_maps.functions.js:26]: undefined is falsy

File: source/js/dg_maps.functions.js [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+
+    /**
+     * The url to the group layer icon.
+     *
+     * @private
+     * @type {string}
+     */
+    this.layerGroupIcon_ = typeof options.layerGroupIcon !== 'undefined' ? options.layerGroupIcon : void 0;
```
Comment (delrueba) [inline source/js/dg_maps.functions.js:34]: see above

File: source/js/dg_maps.functions.js [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+
+    /**
+     * The layer legend.
+     *
+     * @private
+     * @type {Object}
+     */
+    this.legend_ = typeof options.legend !== 'undefined' ? options.legend : {};
```
Comment (delrueba) [inline source/js/dg_maps.functions.js:42]: see above

File: source/js/dg_maps.functions.js [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+    label.innerHTML = layer.get('title');
+    inputWrapper.appendChild(label);
+
+    li.appendChild(inputWrapper);
+
+    // Add layer legend icon(s).
+    // If more than one, add group icon + icon list.
+    // If single, add icon to layer label & skip icon description.
+    var legendData = this.getLegend(layerId);
+    if (legendData.length > 0) {
```
Comment (delrueba) [inline source/js/dg_maps.functions.js:115]:
```suggestion
    if (legendData.length) {
```

File: source/js/dg_maps.functions.js [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+      // Always use the group legend item icon to prepend it to the layer label.
+      var layerIcon = (legendData.length === 1) ? legendData[0] : this.layerGroupIcon_;
+
+      var legendIcon = document.createElement('img');
+      legendIcon.className = 'layer__icon';
+      legendIcon.src = layerIcon.url;
+      legendIcon.alt = layerIcon.label ? layerIcon.label : '';
```
Comment (delrueba) [inline source/js/dg_maps.functions.js:122]: simplify

File: source/js/dg_maps.functions.js (hunk 7 above)
Comment (delrueba) [reply source/js/dg_maps.functions.js:115]: Just noticed you also check (legendData.length > 1) further down.
You can ignore this suggestion in favour of code clarity.

File: source/js/dg_maps.functions.js [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+          legendContainer.appendChild(legendItemContainer);
+
+          var legendItemIcon = document.createElement('img');
+          legendItemIcon.className = 'legend__item__icon';
+          legendItemIcon.src = legendData[j].url;
+          legendItemIcon.alt = legendData[j].label ? legendData[j].label : '';
+          legendItemContainer.appendChild(legendItemIcon);
+
+          var legendItemLabel = document.createElement('span');
+          legendItemIcon.className = 'legend__item__label';
```
Comment (delrueba) [inline source/js/dg_maps.functions.js:142]: Move to above codeblock.

File: source/sass/31-molecules/map/_map.scss [hunk 10]
```diff
@@ -221,5 +310,81 @@ $map-text-color: color('dark-gray');
...
+    padding-left: 1.5rem;
+    font-size: .8rem;
+
+    &__title {
+      display: inline;
+      font-weight: 600;
+
+      &:not(:empty) + .ol-list-features__list__item__description:not(:empty)::before {
+        content: "\2014";
+        padding: 0 1ch;
```
Comment (delrueba) [inline source/sass/31-molecules/map/_map.scss:382]: :trophy:

File: templates/contrib/dg_maps/dg-maps-map-region.html.twig [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+] %}
+
+{% if settings.collapsed %}
+  {% set region_classes = region_classes|merge(['map-region--collapsible']) %}
+{% endif %}
+
+{% set hide_on_mobile = settings.hide_on_mobile ? ' hide-on-mobile' : '' %}
+
+{% if settings.collapsed %}
+  <button type="button" class="map-region-button {{ hide_on_mobile }}" data-toggle-region="{{ name }}" title="{{ "Toggle @name region"|t({ '@name': name }) }}">
```
Comment (delrueba) [inline templates/contrib/dg_maps/dg-maps-map-region.html.twig]: Is this a toggle button?
Would suggest using aria-expanded or aria-pressed.
Only if you're certain that you can manage the states.

**Review by delrueba — CHANGES_REQUESTED** [review]

**Comment by hstandaert** [comment]
@delrueba, I'm going to resolve all your comments regarding the javascript improvements, since this is code that is copy-pasted from the `dg_maps` module. 

Even though your comments are justified, I don't think it makes much sense to optimize third-party code. It'll only make maintenance harder when`dg_maps` is updated.

(Correct me if I'm wrong about this 😆)

**Comment by delrueba** [comment]
> @delrueba, I'm going to resolve all your comments regarding the javascript improvements, since this is code that is copy-pasted from the `dg_maps` module.
> 
> Even though your comments are justified, I don't think it makes much sense to optimize third-party code. It'll only make maintenance harder when`dg_maps` is updated.
> 
> (Correct me if I'm wrong about this laughing)

Well, you can always contribute to dg_maps :stuck_out_tongue:

**Review by delrueba — APPROVED** [review]

**Review by meirege — APPROVED** [review]

**Pull request merged**